rust/connlib/clients/shared/src/control.rs

```diff
@@ -301,10 +304,13 @@ async fn upload(path: PathBuf, url: Url) -> io::Result<()> {
     tracing::info!(path = %path.display(), %url, "Uploading log file");
 
     let file = tokio::fs::File::open(&path).await?;
+
     let response = reqwest::Client::new()
         .put(url)
+        .header(CONTENT_TYPE, "text/plain")
```

Comment:
@AndrewDryga Do you know if GCP will automatically decompress these logfiles upon writing since we set `Content-Type: text/plain` here?

Reply:
It can decompress files when serving them but I think in the bucket itself we will see a gzipped file, so for our use cases - we will have to search through gzips.

Reply:
Gotcha. I was thinking we'd be using these logs to debug historical issues that particular customers/users are facing, in a sort of targeted search. For that use case, I think gzip works great (and saves us on storage costs!).

For live alerting it's probably not the best to store gzips, but I'm not sure we need to send logs up to do that anyway. We'll still have crash reports we can rely on for major OS-level outages (e.g. Apple issues a security update that causes Firezone to crash).

What we may want to do is namespace the log files so we can quickly find them for a particular account-user-client tuple. For that I've opened https://github.com/firezone/firezone/issues/2354